### ISSUE TITLE
[Jessie] OpenLDAP appliance not building

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#418]
````
[ ok ] Starting OpenLDAP: slapd.
Processing triggers for libc-bin (2.19-18) ...
adding new entry "ou=Groups,dc=example,dc=com"

adding new entry "cn=default,ou=Groups,dc=example,dc=com"

adding new entry "ou=Users,dc=example,dc=com"

SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
modifying entry "cn=config"

SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
modifying entry "olcDatabase={1}hdb,cn=config"
ldap_modify: Other (e.g., implementation specific) error (80)
	additional info: duplicate index definition for attr "cn"

Traceback (most recent call last):
  File "/usr/lib/inithooks/bin/openldap.py", line 66, in <module>
    main()
  File "/usr/lib/inithooks/bin/openldap.py", line 63, in main
    system(script, domain, password)
  File "/usr/lib/python2.7/dist-packages/executil.py", line 56, in system
    raise ExecError(command, exitcode)
executil.ExecError: non-zero exitcode (80) for command: /usr/lib/inithooks/bin/openldap-reinit.sh 'example.com'  'turnkey'
/usr/share/fab/product.mk:476: recipe for target 'build/stamps/root.patched' failed
make: *** [build/stamps/root.patched] Error 1
````
Seems to be a config issue. Not sure whether it's because of Jessie or perhaps some more general issue. FWIW I tested building from the v13.0 tag and that builds ok.

**Comment by jstruebel** [comment]
I just took a quick look at this since I had worked with the OpenLDAP appliance a while back. The error is actually occurring in openldap/overlay/usr/lib/inithooks/bin/openldap-reinit.sh on line 93-133. Somehow the olcDbIndex entry for cn already exists in the LDAP database. Possibly one of the other config scripts or possibly now in Jessie that index entry is already defined. Hopefully that helps focus debugging efforts for you. If I get some time later on I can take a look at where it's getting defined, but I'll have to download and setup the v14.0 tkldev first.

**Comment by JedMeister** [comment]
Great thanks. :smile:

**Comment by jstruebel** [comment]
I had some time this weekend to look into this. In Jessie, the default OpenLDAP install added some of the indexes that I had put in which is what caused the error. I fixed the openldap-reinit.sh and created a pull request. Also I found that phpLDAPadmin was broken with Jessie, so I changed the download source to pull the latest master from the github repo. Let me know if I should have it download a specific revision or something else.

One last item. I noticed that when I first install the ISO to a KVM VM that after the firstboot is run I can't load phpLDAPadmin until I restart or reload lighttpd. My suspicion is that the ssl certificate regen is running after lighttpd is started and messing it up somehow, but I wasn't sure where that should get fixed. The errors that I get from the lighttpd error.log are:
...
2015-08-09 04:03:19: (connections.c.305) SSL: 1 error:04075070:rsa routines:RSA_sign:digest too big for rsa key
2015-08-09 04:03:19: (connections.c.305) SSL: 1 error:1409B006:SSL routines:SSL3_SEND_SERVER_KEY_EXCHANGE:EVP lib
2015-08-09 04:03:19: (connections.c.305) SSL: 1 error:140A1175:SSL routines:SSL_BYTES_TO_CIPHER_LIST:inappropriate fallback
2015-08-09 04:03:49: (connections.c.305) SSL: 1 error:04075070:rsa routines:RSA_sign:digest too big for rsa key
2015-08-09 04:03:49: (connections.c.305) SSL: 1 error:1409B006:SSL routines:SSL3_SEND_SERVER_KEY_EXCHANGE:EVP lib
2015-08-09 04:03:49: (connections.c.305) SSL: 1 error:04075070:rsa routines:RSA_sign:digest too big for rsa key
2015-08-09 04:03:49: (connections.c.305) SSL: 1 error:1409B006:SSL routines:SSL3_SEND_SERVER_KEY_EXCHANGE:EVP lib
2015-08-09 04:03:49: (connections.c.305) SSL: 1 error:140A1175:SSL routines:SSL_BYTES_TO_CIPHER_LIST:inappropriate fallback
...
